src/cryptonote_protocol/quorumnet.cpp

```diff
@@ -477,10 +477,13 @@ quorum_vote_t deserialize_vote(const bt_value &v) {
 }
 
 void relay_obligation_votes(void *obj, const std::vector<service_nodes::quorum_vote_t> &votes) {
+    assert(obj);
+    if (!obj) return;
```

Comment:
Either the return or the assert shouldn't be here; either it's okay to call with nullptr, or it's not.  Right now it looks like it could legitimately be, so I think the `assert` shouldn't be here.

Reply:
Actually, no, I think realistically the assert should be here and the caller shouldn't try to call this if not a service node; the assert will correctly abort when compiled in debug mode and it is called as such.